enhancements/machine-api/machine-api-usage-telemetry.md

```diff
@@ -0,0 +1,365 @@
+
+For example, seeing that users are creating many different Machines of varying
+flavors through multiple MachineSets will give us insight into how users would
+prefer to consume cloud resources. Conversely, seeing a high number of Machine
+creations without corresponding MachineSets might indicate that the API is not
+clearly understood, or that Machine resource sizing is less important to users.
+
+**Under-utilized Cluster Autoscaler**
+
+Epics: Engineering Insights, Proactive Support
+
+This metric will help us identify when users have enabled the Cluster Autoscaler
+but are not using it. By collecting data on the number of MachineAutoscalers
+that have been deployed in a cluster correlated with the deployment of the
+Cluster Autoscaler we can identify under-utilization cases. This data
+shows us opportunties to inform users and increase education around usage of
+the autoscaler.
+
+#### Alerts, Metrics, and Telemetry
+
+To fully accomplish the goals proposed in this enhancement several metrics
+will need to be exposed through telemetry. These metrics are:
+
+**These metrics exist today**
+
+* Machine resource count
+    * `mapi_machine_items` - This metric has no labels.
+* MachineSet resource count
+    * `mapi_machineset_items` - This metric has no labels.
+* Cluster Autoscaler scale down count
+    * `cluster_autoscaler_scaled_down_nodes_total` - This metric has one label `reason`,
+    with a cardinality of 3.
+* Cluster Autoscaler scale up count
+    * `cluster_autoscaler_scaled_up_nodes_total` - This metric has no labels.
+
+**These metrics will need to be created**
+
+* MachineAutoscaler resource count
+    * `cao_machineautoscaler_items` - This metric has no labels.
+* MachineHealthCheck resource count
+    * `mapi_machinehealthcheck_items` - This metric has no labels.
+* MachineHealthCheck total nodes covered count
+    * `mapi_machinehealthcheck_total_nodes_count` - This metric has no labels.
+* MachineHealthCheck successful remediations count
+    * `mapi_machinehealthcheck_remediation_success_count` - This metric has no labels.
+
+**Metric series to be exported**
+
+* Total Machine resource count, using `mapi_machine_items` with no labels.
+* Total MachineSet resource count, using `mapi_machineset_items` with no labels.
+* Total Cluster Autoscaler scale down nodes, using `cluster_autoscaler_scaled_down_nodes_total`
+  as a single combined metric of all labels.
+* Total Cluster Autoscaler scale up nodes, using `cluster_autoscaler_scaled_up_nodes_total`
+  with no labels.
+* Total MachineAutoscaler resource count, using `cao_machineautoscaler_items` with no labels.
+* Total MachineHealthCheck resource count, using `mapi_machinehealthcheck_items` with no labels.
+* Total nodes covered by MachineHealthChecks count, using `mapi_machinehealthcheck_total_nodes_count` with no labels.
+* Total remediations completed by MachineHealthChecks count, using `mapi_machinehealthcheck_remediation_success_count` with no labels.
```

Comment:
`_count` is a reserved suffix, best practice is not to use it to not confuse users, you just want something along the lines of `mapi_machinehealthcheck_remediation_successes` (unless this is a true counter, in which case this should be the `_total` suffix instead of `_count`)

Reply:
ack, i will update these. thanks!

Reply:
I should have also been clearer in my first comment here, counters are basically always better if you can model your data as that, and this seems like a perfect use case.